src/libUtils/SafeMath.tpp

```diff
@@ -118,26 +118,12 @@ bool SafeMath<T>::power(const T& base, const T& exponent, T& result) {
 // Now only used for declare constant variable in Constants.cpp
 template <class T>
 T SafeMath<T>::power(const T& base, const T& exponent, bool isCritical) {
-  if (exponent == 0) {
-    return 1;
-  }
-
-  if (exponent < 0) {
-    LOG_GENERAL(WARNING, "Doesn't support pow with negative index");
-    return 0;
-  }
-
-  T ret = base, count = exponent - 1;
-
-  while (count > 0) {
-    if (!SafeMath::mul(ret, base, ret)) {
-      LOG_GENERAL(isCritical ? FATAL : WARNING,
-                  "SafeMath::pow failed ret: " << ret << " base " << base);
-      return ret;
-    }
-    --count;
+  T ret{};
+  if (!SafeMath::power_core(base, exponent, ret)) {
+    LOG_GENERAL(isCritical ? FATAL : WARNING,
```

Comment:
When will be isCritical enabled?

Reply:
This is from previous version of code, if u see, I also dont know why it is there.

Reply:
I think can be removed, all the other safe math function is log warning